Sources/Appwrite/Client.swift

```diff
@@ -530,23 +530,23 @@ open class Client {
             if param is String
                 || param is Int
                 || param is Float
+                || param is Double
                 || param is Bool
                 || param is [String]
                 || param is [Int]
                 || param is [Float]
+                || param is [Double]
                 || param is [Bool]
                 || param is [String: Any]
                 || param is [Int: Any]
                 || param is [Float: Any]
+                || param is [Double: Any]
                 || param is [Bool: Any] {
                 encodedParams[key] = param
-            } else {
-                let value = try! (param as! Encodable).toJson()
-
-                let range = value.index(value.startIndex, offsetBy: 1)..<value.index(value.endIndex, offsetBy: -1)
-                let substring = value[range]
-
```

Comment:
is this expected ?

Reply:
Yep, this was a bug in encoding nested objects when using generics: https://github.com/appwrite/sdk-generator/pull/1004